From 461e0340b5a31ec724914212c9e21debf5630bf3 Mon Sep 17 00:00:00 2001
From: Peter Vary <pvary@cloudera.com>
Date: Thu, 14 Jul 2016 11:27:18 -0500
Subject: [PATCH 0674/1164] CDH-42007 HIVE-14215: Displaying inconsistent CPU
 usage data with MR execution engine (Peter Vary,
 reviewed by Aihua Xu and Sergio Pena)

Change-Id: I82c6d08b5a7f1d52f481d3ea7acd024b3cc4ccc3
---
 .../hive/ql/exec/mr/HadoopJobExecHelper.java       |   25 ++++++++++----------
 1 file changed, 13 insertions(+), 12 deletions(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/exec/mr/HadoopJobExecHelper.java b/ql/src/java/org/apache/hadoop/hive/ql/exec/mr/HadoopJobExecHelper.java
index 6a6593c..fc10efe 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/exec/mr/HadoopJobExecHelper.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/exec/mr/HadoopJobExecHelper.java
@@ -373,6 +373,19 @@ private MapRedStats progress(ExecDriverTaskHandle th) throws IOException {
       reportTime = System.currentTimeMillis();
     }
 
+    Counters ctrs = th.getCounters();
+
+    if (ctrs != null) {
+      Counter counterCpuMsec = ctrs.findCounter("org.apache.hadoop.mapred.Task$Counter",
+              "CPU_MILLISECONDS");
+      if (counterCpuMsec != null) {
+        long newCpuMSec = counterCpuMsec.getValue();
+        if (newCpuMSec > cpuMsec) {
+          cpuMsec = newCpuMSec;
+        }
+      }
+    }
+
     if (cpuMsec > 0) {
       console.printInfo("MapReduce Total cumulative CPU time: "
           + Utilities.formatMsecToStr(cpuMsec));
@@ -380,7 +393,6 @@ private MapRedStats progress(ExecDriverTaskHandle th) throws IOException {
 
     boolean success;
 
-    Counters ctrs = th.getCounters();
     if (fatal) {
       success = false;
     } else {
@@ -398,17 +410,6 @@ private MapRedStats progress(ExecDriverTaskHandle th) throws IOException {
       }
     }
 
-    if (ctrs != null) {
-      Counter counterCpuMsec = ctrs.findCounter("org.apache.hadoop.mapred.Task$Counter",
-          "CPU_MILLISECONDS");
-      if (counterCpuMsec != null) {
-        long newCpuMSec = counterCpuMsec.getValue();
-        if (newCpuMSec > cpuMsec) {
-          cpuMsec = newCpuMSec;
-        }
-      }
-    }
-
     MapRedStats mapRedStats = new MapRedStats(numMap, numReduce, cpuMsec, success, rj.getID().toString());
     mapRedStats.setCounters(ctrs);
 
-- 
1.7.9.5

